lib/trip_dispatcher.rb

```diff
@@ -90,6 +91,60 @@ def load_trips
       trips
     end
 
+    def find_available_driver
+      all_available_drivers = @drivers.select{ |driver| driver.status == :AVAILABLE }
+
+      recent_trips_of_available_drivers = []
+      all_available_drivers.each do |driver|
+        most_recent_trip = driver.trips.first
+        driver.trips.each do |trip|
+          if trip.end_time > most_recent_trip.end_time
+            most_recent_trip = trip
+          end
+          recent_trips_of_available_drivers << most_recent_trip
+        end
+      end
+
+      sorted_trips = recent_trips_of_available_drivers.flatten.sort {|trip_1, trip_2| trip_2.end_time <=> trip_1.end_time } #this puts the oldest correct drivers as the last 5
+
+      oldest_trips = sorted_trips.uniq{ |trip| trip.driver } #using uniq to get rid of dupe trips for drivers
```

Comment:
You shouldn't need to remove duplicate drivers from the list since each driver only has their last trip added to the list.